packages/template/-private/dsl/types.d.ts

```diff
@@ -2,9 +2,21 @@ import { HasContext, HasElement } from '@glint/template/-private/integration';
 
 type Constructor<T> = new (...args: any) => T;
 
-/** Given a tag name, returns an appropriate `Element` subtype. */
-export type ElementForTagName<Name extends string> = Name extends keyof HTMLElementTagNameMap
+type LookupHTMLElement<Name extends string> = Name extends keyof HTMLElementTagNameMap
   ? HTMLElementTagNameMap[Name]
+  : never;
+type LookupSVGElement<Name extends string> = Name extends keyof SVGElementTagNameMap
+  ? SVGElementTagNameMap[Name]
+  : never;
+
+/**
+ * Given a tag name, returns an appropriate `Element` subtype.
+ * NOTE: This will return a union for elements that exist both in HTML and SVG. Technically, this will be too permissive.
+ */
+export type ElementForTagName<Name extends string> = Name extends
+  | keyof HTMLElementTagNameMap
+  | keyof SVGElementTagNameMap
+  ? LookupHTMLElement<Name> | LookupSVGElement<Name>
   : Element;
```

Comment:
Just realized I didn't change this to `HTMLElement | SVGElement` should I still do that?